src/v/cluster/rm_stm.cc

```diff
@@ -103,123 +59,9 @@ raft::replicate_options make_replicate_options() {
-    if (version >= rm_stm::fence_control_record_v1_version) {
-        tx_seq = reflection::adl<model::tx_seq>{}.from(val_reader);
-        transaction_timeout_ms
-          = reflection::adl<std::chrono::milliseconds>{}.from(val_reader);
-    }
-    model::partition_id tm{model::legacy_tm_ntp.tp.partition};
-    if (version >= rm_stm::fence_control_record_version) {
-        tm = reflection::adl<model::partition_id>{}.from(val_reader);
-    }
-
-    auto key_buf = record.release_key();
-    iobuf_parser key_reader(std::move(key_buf));
-    auto batch_type = reflection::adl<model::record_batch_type>{}.from(
-      key_reader);
-    vassert(
-      hdr.type == batch_type,
-      "broken model::record_batch_type::tx_fence batch. expected batch type {} "
-      "got: {}",
-      hdr.type,
-      batch_type);
-    auto p_id = model::producer_id(reflection::adl<int64_t>{}.from(key_reader));
-    vassert(
-      p_id == bid.pid.id,
-      "broken model::record_batch_type::tx_fence batch. expected pid {} got: "
-      "{}",
-      bid.pid.id,
-      p_id);
-    return fence_batch_data{bid, tx_seq, transaction_timeout_ms, tm};
-}
-
-model::record_batch make_fence_batch_v1(
-  model::producer_identity pid,
-  model::tx_seq tx_seq,
-  std::chrono::milliseconds transaction_timeout_ms) {
-    iobuf key;
-    auto pid_id = pid.id;
-    reflection::serialize(key, model::record_batch_type::tx_fence, pid_id);
-
-    iobuf value;
-    reflection::serialize(
-      value,
-      rm_stm::fence_control_record_v1_version,
-      tx_seq,
-      transaction_timeout_ms);
-
-    storage::record_batch_builder builder(
-      model::record_batch_type::tx_fence, model::offset(0));
-    builder.set_producer_identity(pid.id, pid.epoch);
-    builder.set_control_type();
-    builder.add_raw_kv(std::move(key), std::move(value));
-
-    return std::move(builder).build();
-}
-
-model::record_batch make_fence_batch_v2(
-  model::producer_identity pid,
-  model::tx_seq tx_seq,
-  std::chrono::milliseconds transaction_timeout_ms,
-  model::partition_id tm) {
-    iobuf key;
-    auto pid_id = pid.id;
-    reflection::serialize(key, model::record_batch_type::tx_fence, pid_id);
-
-    iobuf value;
-    // the key byte representation must not change because it's used in
-    // compaction
-    reflection::serialize(
-      value,
-      rm_stm::fence_control_record_version,
-      tx_seq,
-      transaction_timeout_ms,
-      tm);
-
-    storage::record_batch_builder builder(
-      model::record_batch_type::tx_fence, model::offset(0));
-    builder.set_producer_identity(pid.id, pid.epoch);
-    builder.set_control_type();
-    builder.add_raw_kv(std::move(key), std::move(value));
-
-    return std::move(builder).build();
-}
-
-model::record_batch rm_stm::make_fence_batch(
-  model::producer_identity pid,
-  model::tx_seq tx_seq,
-  std::chrono::milliseconds transaction_timeout_ms,
-  model::partition_id tm) {
-    if (is_transaction_partitioning()) {
```

Comment:
it looks like the last use of function `is_transaction_partitioning`, should we remove it?